File: Bai_83/main.cpp
#include <iostream>
#include <fstream>
#include <stack>
#include <string>

using namespace std;

int main() {
    ifstream inputFile("BAI2.INP");
    ofstream outputFile("BAI2.OUT");
    
    string n;
    int k;
    
    // Đọc dữ liệu từ file
    if (inputFile.is_open()) {
        inputFile >> n >> k;
        inputFile.close();
    } else {
        cout << "Không thể mở file đầu vào." << endl;
        return 1;
    }
    
    // Xác định vị trí của k chữ số lớn nhất để xoá
    stack<char> st;
    int remaining = k;
    for (char digit : n) {
        while (!st.empty() && remaining > 0 && st.top() < digit) {
            st.pop();
            remaining--;
        }
        st.push(digit);
    }
    
    // Xoá k chữ số
    string result = "";
    while (!st.empty()) {
        if (remaining > 0) {
            st.pop();
            remaining--;
        } else {
            result = st.top() + result;
            st.pop();
        }
    }
    
    // Ghi kết quả vào file
    if (outputFile.is_open()) {
        outputFile << result;
        outputFile.close();
    } else {
        cout << "Không thể mở file đầu ra." << endl;
        return 1;
    }
    
    return 0;
}

File: Bai_83/main.py
# Đọc dữ liệu từ file
with open('BAI2.INP', 'r') as f:
    n = f.readline().strip()
    k = int(f.readline())

# Xác định vị trí của k chữ số lớn nhất để xoá
stack = []
remaining = k
for digit in n:
    while stack and remaining > 0 and stack[-1] < digit:
        stack.pop()
        remaining -= 1
    stack.append(digit)

# Xoá k chữ số
result = ''.join(stack[:len(stack) - remaining])

# Ghi kết quả vào file
with open('BAI2.OUT', 'w') as f:
    f.write(result)
